src/helpers/dom.js

```diff
@@ -145,34 +150,46 @@ export function ensureDomLoaded(func) {
  * @param {function} f - The function to call.
  * @param {function} [f2] - The function to call if the element doesn't exist.
  */
-export function ifElement(query, func, extraFunc) {
+export function ifElement (query, func, extraFunc) {
   ensureDomLoaded(() => {
-    let e = document.querySelector(query);
+    let e = document.querySelector(query)
     if (e) {
-      func(e);
+      func(e)
     } else if (extraFunc) {
-      extraFunc();
+      extraFunc()
     }
-  });
+  })
 }
 /**
- * Will wait for an element to appear on the page and then call a function. This does depend on ensureDomLoaded to make sure it will work. This is used to prevent the script from running if the element doesn't exist.
+ * Will wait for an element to appear on the page and then call a function. This is used to prevent the script from running if the element doesn't exist.
  * @param {string} e - The element to check.
  * @param {function} f - The function to call.
  */
-export function awaitElement(elem, func) {
-  ensureDomLoaded(() => {
-    let time = setInterval(() => {
-      let element = document.querySelector(elem);
-      if (element) {
-        func(element);
-        clearInterval(time);
-      }
-    }, 10);
-    setInterval(() => {
-      clearInterval(time);
-    }, 30000);
-  });
+export function awaitElement (selector, callback) {
+  return new Promise(resolve => {
+    const element = document.querySelector(selector)
+    if (element) {
+      console.log(`[FastForward] Element found: ${element}`)
+      callback(element)
+      resolve(element)
+      return
+    }
+
+    const observer = new MutationObserver(() => {
+      Array.from(document.querySelectorAll(selector)).forEach(el => {
+        console.log(`[FastForward] Element found: ${el}`)
+        callback(el)
+        resolve(el)
+        observer.disconnect()
+      })
+    })
+
+    observer.observe(document.documentElement, {
+      attributes: true,
+      childList: true,
+      subtree: true
+    })
+  })
```

Comment:
Changes to the awaitElement function can be a separate PR, this will probably break a lot of other bypasses

Reply:
Sure, I'll separate it.